src/executor/insert_executor.cpp

```diff
@@ -74,6 +75,14 @@ bool InsertExecutor::DExecute() {
     return false;
   }
 
+  oid_t table_oid = target_table->GetOid();
```

Comment:
My comments may same stupid, why we need lock in insert_executor and update_executor? Are they related to add/drop index?

Reply:
The reason to use locks in add drop index is to prevent index changes during the add/drop index period, so we need to lock operations that modifies the index.

Reply:
You don't need to aquire locks to drop an index.